crates/sui-config/src/swarm.rs

```diff
@@ -216,6 +239,12 @@ impl<'a> FullnodeConfigBuilder<'a> {
         let jsonrpc_server_url = format!("{}:{}", listen_ip, rpc_port);
         let json_rpc_address: SocketAddr = jsonrpc_server_url.parse().unwrap();
 
+        let supported_protocol_versions = match &self.supported_protocol_versions_config {
+            ProtocolVersionsConfig::Default => SupportedProtocolVersions::SYSTEM_DEFAULT,
+            ProtocolVersionsConfig::Global(v) => *v,
+            ProtocolVersionsConfig::PerValidator(func) => func(0, None),
+        };
```

Comment:
Seen this a couple of times, should it just be a function on `ProtocolVersionConfig`?

Reply:
Since the two instances of this code pass different arguments to the callback (to indicate whether it is a validator or a fullnode) I don't see an obvious way to DRY that results in better code. (we could pass the function arguments into a method, but that seems very strange to me).